src/ut/realmmanager_test.cpp

```diff
@@ -149,6 +172,395 @@ TEST_F(RealmmanagerTest, CreateDestroy)
   delete realm_manager;
 }
 
+// Test that the alarm is raised and cleared correctly while attempting to
```

Comment:
These UTs are still too large. This UT covers:
* Peer connection at start of day.
* Alarm is raised if you get fewer than 2 connections because a connection fails due to a realm mismatch.
* No change to alarm state if a peer fails when the alarm is already raised.
* No change to alarm state if a peer succeeds if there are fewer than 2 connections and there's a failed peer.
* Alarm cleared if there's no failed peers and 2 connections
* Alarm raised if you get fewer than 2 connections (not for realm mismatch).
* Peer connection before if the resolver returns no results

For each UT, you really want to pick one thing for it to test. 
* I hope the peer connection at start of day/tear down is already covered by other tests, so you shouldn't be testing it here. Pull this code into start_up/tear_down functions, and don't check the exact parameters (use _).
* Think about what each of the state changes are, and have a UT for each state change.
* You might need to have setup code in your later UTs that's the same/similar to an earlier UT. In this case, make sure that the setup code isn't also a test (it's easy to test too much in UTs as you just pass everything through the same checking function, but this actually leads to fragile UTs).